docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
+For this reason, it has no other runtime to setup than Python itself, can be used to orchestrate your distributed code so you
+can use your Ray or Spark clusters less disruptively within your stack (e.g. sending a function which uses
+Ray over to the head node of the Ray cluster, where the Ray will execute as usual).
+
+<diagram showing Ray calling actors within the cluster, Runhouse calling outside into other processes, VMs,
+Ray clusters, K8s clusters, clouds>
+
+This also fixes some other sharp
+edges with such systems which significantly reduces costs, such as utilizing more than one cluster in an application,
+or sharing a cluster between multiple callers. Is also means the local and remote compute are largely decoupled,
+with no shared runtime which will break if one disconnects or goes down.
+
+*Workflow orchestrators* like Airflow, Prefect, Dagster, Flyte, Metaflow, or Argo can allocate heterogeneous compute
+on the the fly, but act as the runtime itself for the program and only support certain pre-defined and highly
+constrained DAGs. By allocating services Runhouse allows for arbitrary control flow and utilization of remote
+hardware, making Python itself the orchestrator.
+For example, with Runhouse it's easy to allocate small compute to start a training, but if the training fails due to OOM
+restart it with a slightly larger box. Other compute flexibility like multi-region or multi-cloud which other
+orchestrators struggle with are trivial for Runhouse.
+
+<diagram showing Airflow as a sequence of nodes, and Runhouse as a call within Python to an outside node>
+
+Generally, workflow orchestrators are built to be good at monitoring, telemetry, fault-tolerance, and scheduling, so
+we recommend using one strictly for those features and using Runhouse within your pipeline nodes for the heterogeneous
+compute and remote execution. You can also save a lot of money by reusing compute across multiple nodes or reusing
+services across multiple pipelines with Runhouse, which is generally not possible with workflow orchestrators.
+
+<diagram showing an Airflow pipeline calling multiple steps on one cluster, and another pipeline calling a service
+created by the first>
+
+*Serverless frameworks* like Modal or AWS Lambda allow for the allocation of services on the fly, but within a well
+defined sandbox, and not strictly from within regular Python - they require specific pre-packaging or CLI launch
+commands outside Python. Runhouse runs fully in a Python interpreter so it can extend the compute power of practically
+any existing Python application, and allocates services inside your own compute, wherever that may be. We may even
+support serverless systems as compute backends in the future.
+
+<diagram showing Modal creating new services within their own compute, and Runhouse within EC2, K8s, GCP, etc.>
+
+*Infrastructure in code* tools like SkyPilot or Pulumi allocate compute on the fly, but can't utilize it instantly
```

Comment:
A very general nit: Several sentences in this doc follow this format and don't need the comma before "but" (unless the second half is a standalone sentence or a the comma comes from an inserted thought)